packages/api-server/src/convert-tx.ts

```diff
@@ -324,40 +350,56 @@ export async function parseRawTransactionData(
   const args_48_52 = UInt32ToLeBytes(dataByteLength);
   // data
   const args_data = data;
+  let args_7 = to === DEPLOY_TO_ADDRESS ? "0x03" : "0x00";
 
-  let args_7 = "";
+  const isEthNativeTransfer_ = await isEthNativeTransfer(rawTx, rpc);
   let toId: HexNumber | undefined;
   if (to === DEPLOY_TO_ADDRESS) {
-    args_7 = "0x03";
+    toId = gwConfig.accounts.polyjuiceCreator.id;
+  } else if (isEthNativeTransfer_) {
     toId = gwConfig.accounts.polyjuiceCreator.id;
   } else {
-    args_7 = "0x00";
     toId = await getAccountIdByEthAddress(to, rpc);
   }
 
   if (toId == null) {
     throw new Error(`to id not found by address: ${to}`);
   }
 
-  // disable to address is eoa case
-  const toScriptHash = await rpc.getScriptHash(Number(toId));
-  const eoaScriptHash = ethEoaAddressToScriptHash(to);
-  if (toScriptHash === eoaScriptHash) {
-    throw new Error(
-      `to_address can not be EOA address! more info: ${COMPATIBLE_DOCS_URL}`
-    );
+  // When it is a native transfer transaction, we do some modifications to Godwoken transaction:
+  //
+  // We make some convention for native transfer transaction:
+  // - Set `gwTx.to_id = <Polyjuice Creator Account ID>`, to mark the transaction is native transfer
+  // - Append the original `tx.to` address to `gwTx.args`, to pass the recipient information to Polyjuice
+  //
+  // See also:
+  // - https://github.com/nervosnetwork/godwoken/pull/784
+  // - https://github.com/nervosnetwork/godwoken-polyjuice/pull/173
+  let args: HexString;
+  if (isEthNativeTransfer_) {
+    const toAddress = rawTx.to;
+    args =
+      "0x" +
+      args_0_7.slice(2) +
+      args_7.slice(2) +
+      args_8_16.slice(2) +
+      args_16_32.slice(2) +
+      args_32_48.slice(2) +
+      args_48_52.slice(2) +
+      args_data.slice(2) +
+      toAddress.slice(2);
+  } else {
+    args =
+      "0x" +
```

Comment:
Why not reuse this part ? Only append `toAddress` to the end.

Reply:
I don't think it's simpler. It redundant, hard to read and hard to maintain.

Reply:
at first glance, it is hard to know if the first part is the same. I think reusing the part is better too

Reply:
😥
Will adapt the code style soon.

Reply:
Done. 1db0157bf7f1d7b2a4b0f9d2f66ddcedf67b5f1a applies the suggested codes.